pkg/workspace/data_values_pre_processing.go

```diff
@@ -151,30 +178,23 @@ func (o DataValuesPreProcessing) overlay(valuesDoc, newValuesDoc *yamlmeta.Docum
 	return newLeft.(*yamlmeta.DocumentSet).Items[0], nil
 }
 
-func (o DataValuesPreProcessing) overlayValuesOverlays(valuesDoc *yamlmeta.Document) (*yamlmeta.Document, error) {
-	if valuesDoc == nil {
-		// TODO get rid of assumption that data values is a map?
-		valuesDoc = &yamlmeta.Document{
-			Value:    &yamlmeta.Map{},
-			Position: filepos.NewUnknownPosition(),
+func (o DataValuesPreProcessing) annotateWithDefaultMissingOk(doc *yamlmeta.Document) {
+	anns := template.NewAnnotations(doc)
+	_, defaultsSetAlready := anns[yttoverlay.AnnotationMatchChildDefaults]
+	if !defaultsSetAlready {
+		anns[yttoverlay.AnnotationMatchChildDefaults] = template.NodeAnnotation{
+			Kwargs: []starlark.Tuple{{
+				starlark.String(yttoverlay.MatchAnnotationKwargMissingOK),
+				starlark.Bool(true),
+			}},
 		}
 	}
+	doc.SetAnnotations(anns)
+}
 
-	var result *yamlmeta.Document
-
-	// by default return itself
-	result = valuesDoc
-
-	for _, valuesOverlay := range o.valuesOverlays {
-		var err error
-
-		result, err = o.overlay(result, valuesOverlay.Doc)
-		if err != nil {
-			// TODO improve error message?
-			return nil, fmt.Errorf("Overlaying additional data values on top of "+
-				"data values from files (marked as @data/values): %s", err)
-		}
+func (o DataValuesPreProcessing) NewEmptyDataValuesDocument() *yamlmeta.Document {
```

Comment:
seems like this method could be inlined where it's used, or at least made private.

Reply:
... in part because there's no need for any code outside this module to generate an empty Data Values.